### PR TITLE
close #13729

### LINKED ISSUE #13729: approx_top_k

### PR DESCRIPTION
**Pull request by nandu-k01**
no test function in test_general_functions.py

closes #13729

### CODE REVIEW

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Great work so far! Please also add a test for the function in its related test file! I'm not sure I understand what you mean by no `test_function` in `test_general_functions`. `test_function` is a testing helper in the `ivy_tests/test_ivy/helpers` submodule, so you'll have to import this as `helpers` and use something like `helpers.test_function` and so on. Thanks a lot!

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 1]
```diff
@@ -448,3 +448,20 @@ def unstack(value: ivy.Array, axis=0, num=None, name=None):
 @to_ivy_arrays_and_back
 def reverse(tensor, axis, name=None):
     return ivy.flip(tensor, axis=axis)
+
+
+@to_ivy_arrays_and_back
+def approx_top_k(values, k=1, name=None):
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/general_functions.py:454]: also, please match the parameters as per the original tf function, thanks!
ref: https://www.tensorflow.org/api_docs/python/tf/approx_top_k

Comment (nandu-k01) [reply ivy/functional/frontends/tensorflow/general_functions.py:454]: i will add a test function for this and make the requested changes

**Comment by juliagsy** [comment]
you may refer to the docs: https://lets-unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html
or the other tests in the same test file: https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py

**Comment by nandu-k01** [comment]
hey getting a weird error from github while commiting "Git: bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)"

**Comment by juliagsy** [comment]
Hey! Apologies for the delayed response! I've never seen that error before unfortunately, I would suggest that you ask in the respectively Discord channel to get help from the community or search online if people have faced similar issues before! Thanks!

**Comment by nandu-k01** [comment]
Hey, the parameter changes that you requested is ok right?and about the git error still looking for a solution

**Comment by juliagsy** [comment]
Hello! I'll leave a review soonest possible! Thanks!

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 2]
```diff
@@ -461,3 +461,20 @@ def norm(tensor, ord="euclidean", axis=None, keepdims=None, name=None):
...
+
+
+@to_ivy_arrays_and_back
+def approx_top_k(input, k=1, name=None):
+    input = ivy.array(input)
+    batch_size = input.shape[0]
+    num_classes = input.shape[1]
+
+    # Approximate the top k values
+    flattened_values = ivy.reshape(input, [batch_size * num_classes])
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/general_functions.py:473]: just wondering, are the reshapes necessary or can it be detected through the shape and pass `axis=<dim_for_op>` in `ivy.top_k`?

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 3]
```diff
@@ -461,3 +461,20 @@ def norm(tensor, ord="euclidean", axis=None, keepdims=None, name=None):
...
+    _, indices = ivy.top_k(flattened_values, k=k)
+    indices = ivy.reshape(indices, [batch_size, k])
+
+    # Gather the top k indices and values
+    gathered_indices = tf_frontend.gather(input, indices, axis=1)
```
Comment (juliagsy) [inline ivy/functional/frontends/tensorflow/general_functions.py:478]: I think `ivy.gather` should be used instead

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py [hunk 4]
```diff
@@ -1789,3 +1789,29 @@ def test_tensorflow_norm(
...
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
+        k=3,  # Set the k value for testing purposes
+        name="test_approx_top_k",  # Set a name for the test
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1816]: name doesn't need to be set

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py [hunk 5]
```diff
@@ -1789,3 +1789,29 @@ def test_tensorflow_norm(
...
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
+        k=3,  # Set the k value for testing purposes
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1815]: this should be generated through hypothesis in the decorator

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Great work so far! I've left some comments below, please ensure that the tests are passing after the changes and request my review again when it's done! Thanks a lot!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**